docs/src/pages/components/data-grid/demo/FullFeauredDemo.js

```diff
@@ -0,0 +1,282 @@
+      position: 'absolute',
+      display: 'table',
+      border: '2px solid #fff',
+      borderTop: 0,
+      borderLeft: 0,
+      transform: 'rotate(45deg) translate(-50%,-50%)',
+      opacity: 1,
+      transition: 'all .2s cubic-bezier(.12,.4,.29,1.46) .1s',
+      content: '""',
+      top: '50%',
+      left: '39%',
+      width: 5.71428571,
+      height: 9.14285714,
+    },
+    '& .MuiCheckbox-root.MuiCheckbox-indeterminate .MuiIconButton-label:after': {
+      width: 8,
+      height: 8,
+      backgroundColor: '#1890ff',
+      transform: 'none',
+      top: '39%',
+      border: 0,
+    },
+  },
+}));
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: 600,
+    width: '100%',
+    '& .MuiFormGroup-options': {
+      alignItems: 'center',
+      '& fieldset': {
+        padding: theme.spacing(2),
+      },
+      '& button': {
+        marginLeft: theme.spacing(1),
+      },
+    },
+  },
+}));
+
+function SettingsPanel(props) {
+  const { onApply, type, size, theme } = props;
+  const [sizeState, setSize] = React.useState(size);
+  const [typeState, setType] = React.useState(type);
+  const [selectedPaginationValue, setSelectedPaginationValue] = React.useState(-1);
+  const [activeTheme, setActiveTheme] = React.useState(theme);
+
+  const handleSizeChange = React.useCallback((event) => {
+    setSize(Number(event.target.value));
+  }, []);
+
+  const handleDatasetChange = React.useCallback((event) => {
+    setType(event.target.value);
+  }, []);
+
+  const handlePaginationChange = React.useCallback((event) => {
+    setSelectedPaginationValue(event.target.value);
+  }, []);
+
+  const handleThemeChange = React.useCallback((event) => {
+    setActiveTheme(event.target.value);
+  }, []);
+
+  const handleApplyChanges = React.useCallback(() => {
+    onApply({
+      size: sizeState,
+      type: typeState,
+      pagesize: selectedPaginationValue,
+      theme: activeTheme,
+    });
+  }, [sizeState, typeState, selectedPaginationValue, activeTheme, onApply]);
+
+  return (
+    <FormGroup className="MuiFormGroup-options" row>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Dataset</FormLabel>
+        <Select value={typeState} onChange={handleDatasetChange}>
+          <MenuItem value="Employee">Employee</MenuItem>
+          <MenuItem value="Commodity">Commodity</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl size="small" component="fieldset">
+        <FormLabel component="legend">Rows</FormLabel>
+        <Select value={sizeState} onChange={handleSizeChange}>
+          <MenuItem value={100}>100</MenuItem>
+          <MenuItem value={1000}>{Number(1000).toLocaleString()}</MenuItem>
+          <MenuItem value={10000}>{Number(10000).toLocaleString()}</MenuItem>
+          <MenuItem value={100000}>{Number(100000).toLocaleString()}</MenuItem>
+        </Select>
+      </FormControl>
```

Comment:
This select component snippet is wrong. I don't know where it's coming from but no. We don't document this.

Reply:
I took the bar from the `demo-app`. I can rework the snippet to be more polished.

Reply:
The errors I have seen:

1. FormLabel instead of InputLabel
2. Not correct component structure, should be a label
3. Flickering bug in Chrome which can from 2.
4. SSR unstable usage of toLocaleString. But it doesn't seem to really be an issue because we don't format the default option. It's inconsistent but an interesting tradeoff, feel free to ignore this point.

Reply:
Updated